### PR TITLE
Various Fixes for Regressions

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: packages/core/src/view/view.ts [hunk 1]
```diff
@@ -314,12 +314,10 @@ function createViewNodes(view: ViewData) {
 export function checkNoChangesView(view: ViewData) {
   Services.updateDirectives(view, CheckType.CheckNoChanges);
   execEmbeddedViewsAction(view, ViewAction.CheckNoChanges);
-  execQueriesAction(
-      view, NodeFlags.TypeContentQuery, NodeFlags.DynamicQuery, CheckType.CheckNoChanges);
   Services.updateRenderer(view, CheckType.CheckNoChanges);
   execComponentViewsAction(view, ViewAction.CheckNoChanges);
-  execQueriesAction(
-      view, NodeFlags.TypeViewQuery, NodeFlags.DynamicQuery, CheckType.CheckNoChanges);
+  // Note: We don't check queries for changes as we didn't do this in ng2.
```
Comment (vicb) [inline packages/core/src/view/view.ts]: ng2 -> v2.x
ng5 -> v5.x

**Review by vicb — COMMENTED** [review]
`"This would be a breaking change compared to v2"`
I have to read it 3 times to get the meaning, could you change to `"Throwing on query changes would be a breaking change compared to v2"`

**Review by vicb — APPROVED** [review]

File: packages/compiler/src/metadata_resolver.ts [hunk 2]
```diff
@@ -662,7 +663,10 @@ export class CompileMetadataResolver {
   }
 
   getInjectableSummary(type: any): cpl.CompileTypeSummary {
-    return {summaryKind: cpl.CompileSummaryKind.Injectable, type: this._getTypeMetadata(type)};
+    return {
+      summaryKind: cpl.CompileSummaryKind.Injectable,
+      type: this._getTypeMetadata(type, undefined, false)
```
Comment (vicb) [inline packages/compiler/src/metadata_resolver.ts]: undefined -> null

**Comment by vicb** [comment]
LGTM with a few updates and once Travis is green

File: packages/core/src/view/view.ts (hunk 1 above)
Comment (tbosch) [inline packages/core/src/view/view.ts]: Done

File: packages/compiler/src/metadata_resolver.ts (hunk 2 above)
Comment (tbosch) [inline packages/compiler/src/metadata_resolver.ts]: Done

**Review by chuckjaz — APPROVED** [review]

**Review by alexeagle — APPROVED** [review]

**Pull request merged**